src/codegate/api/v1.py

```diff
@@ -665,3 +671,89 @@ async def get_workspace_token_usage(workspace_name: str) -> v1_models.TokenUsage
     except Exception:
         logger.exception("Error while getting messages")
         raise HTTPException(status_code=500, detail="Internal server error")
+
+
+@v1.get("/personas", tags=["Personas"], generate_unique_id_function=uniq_name)
+async def list_personas() -> List[Persona]:
+    """List all personas."""
+    try:
+        personas = await dbreader.get_all_personas()
+        return personas
+    except Exception:
+        logger.exception("Error while getting personas")
+        raise HTTPException(status_code=500, detail="Internal server error")
+
+
+@v1.get("/personas/{persona_name}", tags=["Personas"], generate_unique_id_function=uniq_name)
```

Comment:
Can we use an annotated object to do input validation on `persona_name`? I'd like to restrict the characters that folks can use here to prevent vulnerabilities.